MBBSEmu/HostProcess/MbbsHost.cs

```diff
@@ -91,13 +93,24 @@ public class MbbsHost : IMbbsHost
         /// </summary>
         private readonly Timer _timer;
 
+        /// <summary>
+        ///     Timer to prevent main loop from executing while cleanup is running
+        /// </summary>
+        private readonly Timer _nightlyCleanupTimer;
+
+        /// <summary>
+        ///     Flag to prevent main loop from executing while cleanup is running
+        /// </summary>
+        private bool _performingNightlyCleanup;
+        
         /// <summary>
         ///     Flag that controls whether the main loop will perform a nightly cleanup
         /// </summary>
         private bool _performCleanup = false;
 
         private readonly IGlobalCache _globalCache;
         private readonly IFileUtility _fileUtility;
+        private List<ModuleConfiguration> _moduleConfigurations;
```

Comment:
`readonly` too if possible

Reply:
If I add readonly it won't compile because of the below code:

```
           //Save module configurations for cleanup
            _moduleConfigurations = moduleConfigurations;
```